sdk/storage/Azure.Storage.DataMovement/src/Shared/TransferManager.cs

```diff
@@ -260,6 +262,38 @@ public virtual Task<bool> TryPauseTransferAsync(string transferId, CancellationT
             }
         }
 
+        /// <summary>
+        /// Gets the current transfers stored in the <see cref="TransferManager"/>.
+        /// </summary>
+        /// <param name="filterByStatus">
+        /// If specified, the returned list of transfers will have only have the transfers
+        /// of which match the status specified.
+        ///
+        /// If not specified or specified to <see cref="StorageTransferStatus.None"/>,
+        /// all transfers will be returned regardless of status.
+        /// </param>
+        /// <returns></returns>
+        public virtual async IAsyncEnumerable<DataTransfer> GetTransfersAsync(
+            StorageTransferStatus[] filterByStatus = default)
+        {
+            await SetDataTransfers().ConfigureAwait(false);
+            IEnumerable<DataTransfer> totalTransfers;
+            if (filterByStatus == default)
+            {
+                totalTransfers = _dataTransfers.Select(d => d.Value);
+            }
+            else
+            {
+                totalTransfers = _dataTransfers
+                    .Select(d => d.Value)
+                    .Where(x => filterByStatus.Contains(x.TransferStatus)).ToList();
+            }
+            foreach (DataTransfer transfer in totalTransfers)
```

Comment:
Is there a way to just return the `IEnumerable` we have as an `IAsyncEnumerable`? Seems a bit odd to loop through the enumerable we just made only to yield return its results.

Reply:
Actually in C#, you cannot directly return an instance of `IAsyncEnumerable` without using a `yield return` or one of the alternatives. :(
 
 I actually did try using the `ToAsyncEnumerable` method from `System.Linq.Async` but then I ran to the above reasoning above.